app/controllers/mass_rollouts_controller.rb

```diff
@@ -9,37 +9,48 @@ def new
 
   # No more than one stage, per project, per deploy_group
   # Note: you can call this multiple times, and it will create missing stages, but no redundant stages.
+  # TODO: only create selected list like #deploy does
   def create
     stages_created = create_all_stages.compact
 
     redirect_to deploy_group, notice: "Created #{stages_created.length} Stages"
   end
 
-  def deploy
-    stages_to_deploy = deploy_group.stages.to_a
-
-    case params[:status].presence
-    when nil # rubocop:disable Lint/EmptyWhen
-      # all
-    when 'successful'
-      stages_to_deploy.select!(&:last_successful_deploy)
-    when 'missing'
-      stages_to_deploy.reject!(&:last_successful_deploy)
+  def review_deploy
+    if stage_ids = params[:stage_ids]
+      @stages = Stage.find(stage_ids)
     else
-      return unsupported_option(:status)
-    end
+      @stages = deploy_group.stages.to_a
+
+      case params[:status].presence
+      when nil # rubocop:disable Lint/EmptyWhen
+        # all
+      when 'successful'
+        @stages.select!(&:last_successful_deploy)
+      when 'missing'
+        @stages.reject!(&:last_successful_deploy)
+      else
+        return unsupported_option(:status)
+      end
 
-    case params[:kubernetes].to_s.presence
-    when nil # rubocop:disable Lint/EmptyWhen
-      # all
-    when 'true'
-      stages_to_deploy.select!(&:kubernetes?)
-    when 'false'
-      stages_to_deploy.reject!(&:kubernetes?)
-    else
-      return unsupported_option(:kubernetes)
+      case params[:kubernetes].to_s.presence
+      when nil # rubocop:disable Lint/EmptyWhen
+        # all
+      when 'true'
+        @stages.select!(&:kubernetes?)
+      when 'false'
+        @stages.reject!(&:kubernetes?)
+      else
+        return unsupported_option(:kubernetes)
+      end
     end
+  end
 
+  # TODO: show when deploys are waiting for buddy
+  # TODO: show when deploys were invalid
+  # TODO: show when stage had no reference
+  def deploy
+    stages_to_deploy = Stage.find(params.require(:stage_ids))
     deploy_references = stages_to_deploy.map do |stage|
```

Comment:
Would be nice to do this in the review stage as well for the UI, show the reference and show a list of "unable to deploy" projects as well.

Reply:
yeah, I was a bit scared of this one since if you keep the page open too long you end up deploying a old release ... so this could only be a preview or so

Reply:
which projects/stages get deployed will only be a preview too.. alternatively you could send the references with the POST request..